components/org.wso2.carbon.status.dashboard.core/src/main/java/org/wso2/carbon/status/dashboard/core/impl/MonitoringApiServiceImpl.java

```diff
@@ -2174,15 +2260,19 @@ public Response getHASiddhiApps(String username) throws NotFoundException {
                         feign.Response resourceResponse = WorkerServiceFactory.getWorkerHttpsClient(
                                 PROTOCOL + generateURLHostPort(manager.getHost(), String.valueOf(manager
                                         .getPort())), this.getUsername(), this.getPassword()).getClusterNodeDetails();
-                        if (resourceResponse.status() == 200) {
-                            Reader inputStream = resourceResponse.body().asReader();
-                            List<ResourceClusterInfo> clusterInfos = gson.fromJson(
-                                    inputStream, new TypeToken<List<ResourceClusterInfo>>() {
-                                    }.getType());
-                            for (ResourceClusterInfo clusterInfo : clusterInfos) {
-                                String workerId = generateWorkerKey(clusterInfo.getHttps_host(), clusterInfo
-                                        .getHttps_port());
-                                ResourceClusteredWorkerNode.add(workerId);
+                        if (resourceResponse == null) {
+                            logger.error("Requested response is null");
```

Comment:
Shouldn't we re-try?

Reply:
even though if we are re-trying, always that response is null